File: dto/dtos/StateName.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using dto.Paging;

namespace dto.dtos
{
    public class StateNameRes
    {
        public int StateNameId { get; set; }

        public int? LanguageId { get; set; }
        public LanguageRes? Language { get; set; }

        public int? StateId { get; set; }
        public StateRes? State { get; set; }

        public string? Name { get; set; }
    }
    public class StateNameReqEdit
    {
        [Required]
        public int? LanguageId { get; set; }
        
        [Required]
        public int? StateId { get; set; }
        
        [Required]
        public string? Name { get; set; }
    }

    public class StateNameReqSearch : PagedReq
    {
        [Required]
        public int? StateId { get; set; }
    }
}

File: admin/src/dtos/StateName.ts
import Common from "../utility/Common";
import { LanguageRes } from "./Language";
import { PagedReq } from "./PagedReq";
import { StateRes } from "./State";

export interface StateNameRes {
  stateNameId?: string;
  name?: string;

  languageId?: string;
  language?: LanguageRes;
  stateId?: string;
  state?: StateRes;
}

export class StateNameReqEdit {
  languageId?: string = "";
  stateId?: string = "";
  name?: string = "";

  constructor(languageId?: string, stateId?: string) {
    this.languageId = languageId;
    this.stateId = stateId;
  }
}

export class StateNameReqSearch extends PagedReq {
  stateId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {stateId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.stateId = stateId;
  }
}